pkg/apis/autoscaling/validation/validation.go

```diff
@@ -342,6 +381,36 @@ func validatePodsSource(src *autoscaling.PodsMetricSource, fldPath *field.Path)
 	return allErrs
 }
 
+func validateContainerResourceSource(src *autoscaling.ContainerResourceMetricSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(src.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "must specify a resource name"))
+	} else {
+		if !helper.IsStandardContainerResourceName(string(src.Name)) {
```

Comment:
Right now the resource field for the regular resource type does not have any validation in it. So users can specify even `ephemeral-storage` and `hugepage-` resources for scaling. However `metrics-server` which is used by the HPA controller does not support those resources at the moment. But if they were, the HPA controller would support them out of the box because the resource name is not validated. So should I now restrict which resources can be specified(limit to cpu and memory) or should I be a bit more permissive and allow all container resources.

Reply:
cc @josephburnett

Reply:
I think [replica_calculator.go](https://github.com/kubernetes/kubernetes/blob/e39bdd4ac123217a101ae5099938a566e00059b0/pkg/controller/podautoscaler/metrics/rest_metrics_client.go#L82) is pretty generic all the way through.  So we shouldn't add this validation.

Reply:
@josephburnett @liggitt In that case the current validation is correct. So I will leave this as is.

Reply:
> In that case the current validation is correct

are you sure?  this validation is narrower than `validateContainerResourceName`, so it would disallow resources a container could request, right?

Reply:
I misunderstood your first comment. You're right this is more restrictive. So now I have 3 options:
1. Copy the `validateContainerResourceName` function. 
2. Turn the `validateContainerResourceName` into a public function and add tests.
3. Remove the validation to keep it the same as validation for the resource name in the `ResourceMetricSource`. 

I prefer option 2. WDYT?

Reply:
2 sounds good to me

Reply:
@liggitt I've made changes. Can you take another look.